Apps/WebClient/src/NewClientApp/src/components/private/timeline/comment/CommentSectionComponent.vue

```diff
@@ -0,0 +1,125 @@
+<script setup lang="ts">
+import { computed, onMounted, ref } from "vue";
+
+import HgButtonComponent from "@/components/common/HgButtonComponent.vue";
+import AddCommentComponent from "@/components/private/timeline/comment/AddCommentComponent.vue";
+import CommentComponent from "@/components/private/timeline/comment/CommentComponent.vue";
+import { CommentEntryType } from "@/constants/commentEntryType";
+import { entryTypeMap } from "@/constants/entryType";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { DateWrapper } from "@/models/dateWrapper";
+import TimelineEntry from "@/models/timeline/timelineEntry";
+import { UserComment } from "@/models/userComment";
+import { ILogger } from "@/services/interfaces";
+import { useCommentStore } from "@/stores/comment";
+import { useUserStore } from "@/stores/user";
+
+interface Props {
+    parentEntry: TimelineEntry;
+    isMobileDetails?: boolean;
+    visible?: boolean;
+}
+const props = withDefaults(defineProps<Props>(), {
+    isMobileDetails: false,
+    visible: false,
+});
+
+const newComment: UserComment = {
+    id: "",
+    text: "",
+    parentEntryId: props.parentEntry.id,
+    entryTypeCode:
+        entryTypeMap.get(props.parentEntry.type)?.commentType ??
+        CommentEntryType.None,
+    userProfileId: "",
+    createdDateTime: new DateWrapper().toISODate(),
+    version: 0,
+};
+
+const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+const userStore = useUserStore();
+const commentStore = useCommentStore();
+
+const showComments = ref(false);
+const isLoadingComments = ref(false);
+
+const commentButtonText = computed(() =>
+    commentCount.value === 1 ? "1 Comment" : `${commentCount.value} Comments`
+);
+const commentCount = computed(() => props.parentEntry.comments?.length ?? 0);
+
+onMounted(() => {
+    // Some comments don't have entry type. This code updates them if they don't.
+    const commentsToUpdate: UserComment[] = [];
+    if (props.parentEntry.comments !== null) {
+        props.parentEntry.comments.forEach((x) => {
+            if (x.entryTypeCode === CommentEntryType.None) {
+                x.entryTypeCode =
+                    entryTypeMap.get(props.parentEntry.type)?.commentType ??
+                    CommentEntryType.None;
+                x.updatedBy = "System_Backfill";
+                commentsToUpdate.push(x);
+            }
+        });
+    }
+
+    commentsToUpdate.forEach((x) => {
+        logger.info("Updating comment " + x.id);
+        commentStore.updateComment(userStore.user.hdid, x);
+    });
+});
```

Comment:
This can happen immediately (on create) rather than `onMounted`, since nothing relies on the component being rendered.

Also, [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loops [are preferred](https://thecodebarbarian.com/for-vs-for-each-vs-for-in-vs-for-of-in-javascript) over `forEach()` calls.

Reply:
I agree to the for...of loop, I'll update that, I'm a little concerned about the on create though, cause we have a bunch of timeline components being loaded simultaneously, I'm concerned that we would have some "intense" work, but this requires a few small tests, I'll let you know what I find.